broker/provision_endpoint.go

```diff
@@ -119,3 +119,62 @@ func (bkr *Broker) assertProvisionPrecondition(instanceID structs.ClusterID, fea
 
 	return bkr.scheduler.VerifyClusterFeatures(features)
 }
+
```

Comment:
somehow feel this whole section belongs in its own context... not really part of 'broker'.